tasks/build-monthly-updates-as-json.js

```diff
@@ -0,0 +1,121 @@
+        .pipe(gulp.dest(fileDirectoryDestination));
+}
+
+function parseMarkdown(file) {
+    const env = {};
+    md.render(file.contents.toString(), env);
+    file.title = env.title;
+    file.anchorTitle = env.anchorTitle;
+    file.description = env.description;
+
+    return;
+}
+
+/**
+ * Return the first h1 title in the markdown file.
+ *
+ * Inspired by https://github.com/GerHobbelt/markdown-it-title/blob/master/index.js
+ */
+function parseTitleFromMarkdown(...args) {
+    const [ tokens, idx, , env, self ] = args;
+    let filteredTokens = tokens.filter(t => t.content.indexOf(':::') === -1);
+
+    if (!env.title && filteredTokens[idx].tag === 'h1') {
+        env.title = filteredTokens[idx + 1].content;
+        env.anchorTitle = filteredTokens[idx].attrs[0][1];
+    }
+
+    return self.renderToken(...args);
+};
+
+/**
+ * Return the description of the update feature.
+ *
+ * This description is the content of the paragraph just after the h1 title.
+ * If there is no paragraph after this title, the description is null.
+ */
+function parseDescriptionFromMarkdown(...args) {
+    const [ tokens, idx, , env, self ] = args;
+    let filteredTokens = tokens.filter(t => t.content.indexOf(':::') === -1);
+
+    const firsth1TitleIndex =  filteredTokens.findIndex((token) => token.tag === 'h1' && token.type === 'heading_close');
+    const tokensWithouth1 = filteredTokens.slice(firsth1TitleIndex);
+
+    const firsth2TitleIndex = tokensWithouth1.findIndex((token) => token.tag === 'h2');
+    const descriptionIndex = tokensWithouth1.findIndex((token) => token.type === 'inline');
+
+    if (descriptionIndex === -1) {
+        env.description = null;
+    } else if (firsth2TitleIndex === -1) {
+        env.description = tokensWithouth1[descriptionIndex].content;
+    } else if (descriptionIndex < firsth2TitleIndex) {
+        env.description = tokensWithouth1[descriptionIndex].content;
+    } else {
+        env.description = null;
+    }
+
+    return self.renderToken(...args);
+};
+
+
+function generateJson() {
+    return through((file, enc, cb) => {
+        // hardcoded to the 5th day of the month
+        let directoryName = path.basename(path.dirname(file.path));
+        let startDate = directoryName.replace('-', '/') + '/05';
+        let link = 'https://help.akeneo.com/pim/serenity/updates/' + directoryName + '.html#' + file.anchorTitle;
+
+        let defaultValues = {
+            'pim_announcement_img': null,
+            'pim_announcement_alt_img': null,
+            'pim_announcement_audience': [],
+        };
+
+        let data = {...defaultValues, ...file.fm };
+
+        let content = JSON.stringify({
+            'startDate': startDate,
+            'description': file.description,
+            'img': data['pim_announcement_img'],
+            'imgAlt': data['pim_announcement_alt_img'],
+            'version': data['pim_announcement_audience'],
```

Comment:
@NolwennP  Named it editions and it was an array in the json mocked in step 2. Maybe better than version and audience. What do you think about it ?

Reply:
yes you're right, I will modify it :+1:

Reply:
Ok seen with Nono, she prefers to keep it like that. To be honest, this is the json of the backend, not sure we will have this format in the output of the backend.

Actually, there is no reason to send the version to the PIM (it's the PIM which send the version). So we can do whatever we want. I will adjust it according it to the needs (I'm not even sure to keep an array).